keyboards/tada68/keymaps/bazooka/keymap.c

```diff
@@ -0,0 +1,189 @@
+  _______,_______,_______,_______,_______,_______,_______,_______,KC_MSTP,KC_MPLY,KC_MPRV,KC_MNXT,        _______,KC_HOME, \
+          _______,BL_TOGG, BL_DEC, BL_INC,_______,_______,_______,MOU_TOG,KC_MUTE,KC_VOLD,KC_VOLU,_______,KC_PGUP,KC_END , \
+  _______,WIN_LCK,_______,                        _______,                _______,_______,_______,KC_HOME,KC_PGDN,KC_END),
+
+  /* Keymap _MOUSE: Mouse Navigation Layer
+   * ,----------------------------------------------------------------.
+   * |   |   |   |   |   |   |   |   |   |   |   |   |   |       |    |
+   * |----------------------------------------------------------------|
+   * |     |   |   |   |   |   |   |   |   |   |   |   |   |     |    |
+   * |----------------------------------------------------------------|
+   * |      |   |   |   |   |   |   |   |   |   |   |   |   McR  |    |
+   * |----------------------------------------------------------------|
+   * |        |   |   |   |   |   |   |   |   |   |   |  McL |MsU|    |
+   * |----------------------------------------------------------------|
+   * |    |    |    |                       |   |   |    |MsL|MsD|MsR |
+   * `----------------------------------------------------------------'
+   */
+[_MOUSE] = LAYOUT_ansi(
+  _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+  _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+  _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,        KC_BTN2,_______, \
+          _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,KC_BTN1,KC_MS_U,_______, \
+  _______,_______,_______,                        _______,                _______,_______,_______,KC_MS_L,KC_MS_D,KC_MS_R),
+
+};
+
+#define SET_BIT(var, pos) (var |= (1UL << pos))
+#define CLEAR_BIT(var, pos) (var &= ~(1UL << pos))
+#define TOGGLE_BIT(var, pos) (var ^= (1UL << pos))
+#define CHECK_BIT(var, pos) (var & (1U << pos)) 
+
+static uint8_t keyboard_state = 0;
+// bit 0 = function layer status (on/off)
+// bit 1 = mouse navigation layer state
+// bit 2 = windows key lock state
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch(keycode) {
+    // acts like a momentary function layer press
+    case FUN_LAY:
+      if (record->event.pressed) { 
+        layer_on(_FUNC);
+        if (CHECK_BIT(keyboard_state, 1)) {
+          layer_on(_MOUSE);
+        } 
+        else {
+          layer_off(_MOUSE);
+        }
+        // turns on function layer status
+        SET_BIT(keyboard_state, 0);
+      }
+      // key released
+      else {
+        layer_off(_FUNC);
+        layer_off(_MOUSE);
+        // turns off function layer status
+        CLEAR_BIT(keyboard_state, 0);
+      }
+      break;
+
+    case MAC_TOG:
```

Comment:
You could just use `TG(_MAC)` to do this. 

https://docs.qmk.fm/#/feature_advanced_keycodes?id=switching-and-toggling-layers